packages/server/src/api/students.rs

```diff
@@ -61,8 +61,13 @@ pub async fn update_catalog(
 ) -> Result<HttpResponse, AppError> {
     let obj_id = bson::oid::ObjectId::from_str(&catalog_id)?;
     let catalog = db.get::<Catalog>(&obj_id).await?;
-    user.details.catalog = Some(DisplayCatalog::from(catalog));
-    user.details.modified = true;
+
+    // Courses that are not modified, not completed, and don't have a semester should be removed,
+    // since they were not added by the user and they are irrelevant to the new catalog
+    user.details
+        .degree_status
+        .course_statuses
+        .retain(|cs| cs.modified || cs.completed() || cs.semester.is_some());
```

Comment:
Why did you add it here and not updated the `remove_courses_added_by_algorithm` in `preprocessing.rs`?
https://github.com/sogrim/technion-sogrim/blob/master/packages/server/src/core/degree_status/preprocessing.rs

We call this function anyway whenever we run the compute.

Reply:
Since we set the state of the courses a [few lines after](https://github.com/sogrim/technion-sogrim/blob/6ecf6c0a13849aa89c25fb7a78c37cb6e60e29f6/packages/server/src/api/students.rs#L79), calling this in `preprocessing.rs` would not work because there would be no way to [distinguish algorithm-added courses from user-added courses](https://github.com/sogrim/technion-sogrim/blob/6ecf6c0a13849aa89c25fb7a78c37cb6e60e29f6/packages/server/src/core/degree_status/preprocessing.rs#L27).

It does feel like this feature needs better design so I am open to suggestions..

Reply:
My wondering is as follows:
I am not sure why in line 76 we set the modified of the course to be false? if the user updated the grade of a course, or did any change it makes sense it keep it as true. the only field we want to reset is the type which is invalid in the new catalog's context.
Now, given that we remove this line, the algorithm should work, because we keep only courses which are modified, completed or have a semester; and we don't update any of those fields.

WDYT?

Reply:
My wondering is as follows:
Why in line 76 we set `modified` to false? if the user updated the course it makes sense to keep it modifed.
The only field we need to change (even if the course is modified) is type which is invalid in the new catalog's context.

Now, given that we delete this line, update the relevant lines in `preprocessin.rs` should work because we keep course which are modified, have a semester or completed, and we do not change any of those fields.

WDYT?